### ISSUE TITLE
Excess properties check does not occur for object literals in conditional expressions

### ISSUE THREAD

**Issue by Knagis** [Microsoft/TypeScript#12938]
**TypeScript Version:**  2.1.4 / 2.1.4-insiders.20161201

**Code**

[Code on playground](https://www.typescriptlang.org/play/#src=interface%20IFoo%20%7B%0D%0A%20%20%20%20foo%3F%3A%20number%3B%0D%0A%20%20%20%20bar%3F%3A%20number%3B%0D%0A%7D%0D%0A%0D%0Alet%20a%3A%20IFoo%20%3D%20%7B%0D%0A%20%20%20%20foo%3A%201%2C%0D%0A%20%20%20%20asd%3A%201%20%2F%2F%20this%20line%20shows%20error%0D%0A%7D%3B%0D%0A%0D%0Alet%20b%20%3D%20true%3B%0D%0Alet%20c%3A%20IFoo%20%3D%0D%0A%20%20%20%20b%20%3F%20%7B%0D%0A%20%20%20%20%20%20%20%20asd%3A%202%20%2F%2F%20this%20line%20shows%20error%0D%0A%20%20%20%20%7D%20%3A%20%7B%0D%0A%20%20%20%20%20%20%20%20foo%3A%201%2C%0D%0A%20%20%20%20%20%20%20%20asd%3A%202%20%2F%2F%20this%20line%20does%20NOT%20show%20an%20error%0D%0A%20%20%20%20%7D)

```ts
interface IFoo {
    foo?: number;
    bar?: number;
}

let a: IFoo = {
    foo: 1,
    asd: 2 // this line shows error
};

let b = true;
let c: IFoo =
    b ? {
        asd: 2 // this line shows error
    } : {
        foo: 1,
        asd: 2 // this line does NOT show an error
    }
```

Also this [version](https://www.typescriptlang.org/play/#src=interface%20IFoo%20%7B%0D%0A%20%20%20%20foo%3F%3A%20number%3B%0D%0A%20%20%20%20bar%3F%3A%20number%3B%0D%0A%7D%0D%0A%0D%0Alet%20a%3A%20IFoo%20%3D%20%7B%0D%0A%20%20%20%20foo%3A%201%2C%0D%0A%20%20%20%20asd%3A%201%20%2F%2F%20this%20line%20shows%20error%0D%0A%7D%3B%0D%0A%0D%0Alet%20b%20%3D%20true%3B%0D%0Alet%20c%3A%20IFoo%20%3D%0D%0A%20%20%20%20b%20%3F%20%7B%0D%0A%20%20%20%20%20%20%20%20asd%3A%202%20%2F%2F%20this%20line%20shows%20error%0D%0A%20%20%20%20%7D%20%3A%20%7B%0D%0A%20%20%20%20%20%20%20%20foo%3A%201%2C%0D%0A%20%20%20%20%20%20%20%20asd%3A%202%20%2F%2F%20this%20line%20does%20NOT%20show%20an%20erroor%0D%0A%20%20%20%20%7D#src=interface%20IFoo%20%7B%0D%0A%20%20%20%20foo%3F%3A%20number%3B%0D%0A%20%20%20%20bar%3F%3A%20number%3B%0D%0A%7D%0D%0A%0D%0Alet%20a%3A%20IFoo%20%3D%20%7B%0D%0A%20%20%20%20foo%3A%201%2C%0D%0A%20%20%20%20asd%3A%201%20%2F%2F%20this%20line%20shows%20error%0D%0A%7D%3B%0D%0A%0D%0Alet%20b%20%3D%20true%3B%0D%0Alet%20c%3A%20IFoo%20%3D%0D%0A%20%20%20%20b%20%3F%20%7B%0D%0A%20%20%20%20%7D%20%3A%20%7B%0D%0A%20%20%20%20%20%20%20%20foo%3A%201%2C%0D%0A%20%20%20%20%20%20%20%20asd%3A%202%20%2F%2F%20this%20line%20does%20NOT%20show%20an%20error%0D%0A%20%20%20%20%7D) fails:

```ts
let c: IFoo =
    b ? {
    } : {
        foo: 1,
        asd: 2 // this line does NOT show an error
    }
```

**Expected behavior:**
All `asd:2` assignments should show an error

**Actual behavior:**
Only the first two show the error, the last one does not.

**Comment by mhegazy** [comment]
This is another manifestation of https://github.com/Microsoft/TypeScript/issues/12745. the result of the conditional operator is a union type, and thus the excess property check is ignored.
